### PR TITLE
Feature/presentation layer

### PR DESCRIPTION
**Pull request by cooperative0zero**
Presentation layer has been developed: controllers with minimal logic. Controllers call the appropriate services and return the result to the client.

### CODE REVIEW

File: src/main/java/com/example/practice_modsen_shop/dto/ProductRequestTo.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package com.example.practice_modsen_shop.dto;
+
+import lombok.Data;
+
```
Comment (igortitov2004) [inline src/main/java/com/example/practice_modsen_shop/dto/ProductRequestTo.java]: Мб тут вместо конструкторов использовать аннотации @NoArgsConstructor и @AllArgsConstructor

Comment (cooperative0zero) [reply src/main/java/com/example/practice_modsen_shop/dto/ProductRequestTo.java]: Согласен, исправил.

**Review by FedorDevelopmer — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -54,6 +65,17 @@
             <artifactId>spring-security-test</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (egorzhurov) [inline pom.xml:63]: remove it

File: pom.xml [hunk 3]
```diff
@@ -44,6 +50,11 @@
             <artifactId>lombok</artifactId>
             <optional>true</optional>
         </dependency>
+        <dependency>
```
Comment (egorzhurov) [inline pom.xml]: remove it

File: src/main/java/com/example/practice_modsen_shop/controllers/CategoryController.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import com.example.practice_modsen_shop.services.ICrudService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/categories")
```
Comment (egorzhurov) [inline src/main/java/com/example/practice_modsen_shop/controllers/CategoryController.java]: @RequestMapping("/api/categories")

File: src/main/java/com/example/practice_modsen_shop/controllers/CategoryController.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+public class CategoryController {
+    @Autowired
+    ICrudService<CategoryRequestTo, CategoryResponseTo> categoryService;
```
Comment (egorzhurov) [inline src/main/java/com/example/practice_modsen_shop/controllers/CategoryController.java]: @RequiredArgsConstructor
private final CategoryService categoryService;

File: src/main/java/com/example/practice_modsen_shop/controllers/CategoryController.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+    @GetMapping
+    public ResponseEntity<List<CategoryResponseTo>> getAll(@RequestParam(required = false, defaultValue = "0") Integer pageNumber,
+                                                           @RequestParam(required = false, defaultValue = "10") Integer pageSize,
+                                                           @RequestParam(required = false, defaultValue = "id") String sortBy,
+                                                           @RequestParam(required = false, defaultValue = "desc") String sortOrder) {
+        return ResponseEntity.status(HttpStatus.OK).body(categoryService.getAll(pageNumber, pageSize, sortBy, sortOrder));
```
Comment (egorzhurov) [inline src/main/java/com/example/practice_modsen_shop/controllers/CategoryController.java]: return new ResponseEntity<>(product, HttpStatus.OK);

File: src/main/java/com/example/practice_modsen_shop/controllers/CategoryController.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
+package com.example.practice_modsen_shop.controllers;
```
Comment (egorzhurov) [inline src/main/java/com/example/practice_modsen_shop/controllers/CategoryController.java]: rename controllers to controller

File: src/main/java/com/example/practice_modsen_shop/dto/CategoryRequestTo.java [hunk 8]
```diff
@@ -0,0 +1,13 @@
+package com.example.practice_modsen_shop.dto;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class CategoryRequestTo {
```
Comment (egorzhurov) [inline src/main/java/com/example/practice_modsen_shop/dto/CategoryRequestTo.java]: remove To

File: src/main/java/com/example/practice_modsen_shop/dto/OrderItemRequestTo.java [hunk 9]
```diff
@@ -0,0 +1,19 @@
+package com.example.practice_modsen_shop.dto;
+
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@NoArgsConstructor
+@Getter
```
Comment (egorzhurov) [inline src/main/java/com/example/practice_modsen_shop/dto/OrderItemRequestTo.java]: @Data

File: src/main/java/com/example/practice_modsen_shop/dto/mappers/CategoryMapper.java [hunk 10]
```diff
@@ -0,0 +1,11 @@
+package com.example.practice_modsen_shop.dto.mappers;
+
+import org.mapstruct.Mapper;
+
+@Mapper(componentModel="spring")
+public interface CategoryMapper {
```
Comment (egorzhurov) [inline src/main/java/com/example/practice_modsen_shop/dto/mappers/CategoryMapper.java]: Remove it

File: src/main/java/com/example/practice_modsen_shop/services/OrderItemService.java [hunk 11]
```diff
@@ -0,0 +1,35 @@
+package com.example.practice_modsen_shop.services;
+
+import com.example.practice_modsen_shop.dto.OrderItemRequestTo;
+import com.example.practice_modsen_shop.dto.OrderItemResponseTo;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+@Component
```
Comment (egorzhurov) [inline src/main/java/com/example/practice_modsen_shop/services/OrderItemService.java]: @Service

File: src/test/java/com/example/practice_modsen_shop/CategoryControllerTest.java [hunk 12]
```diff
@@ -0,0 +1,116 @@
+package com.example.practice_modsen_shop;
```
Comment (egorzhurov) [inline src/test/java/com/example/practice_modsen_shop/CategoryControllerTest.java]: add controller package

File: src/test/java/com/example/practice_modsen_shop/CategoryControllerTest.java [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+        CategoryResponseTo categoryResponseTo1 = new CategoryResponseTo(1L, "2");
+        CategoryResponseTo categoryResponseTo2 = new CategoryResponseTo(7L, "6");
+
+        categoryResponseToList.add(categoryResponseTo1);
+        categoryResponseToList.add(categoryResponseTo2);
+    }
+
+    @Test
+    @WithMockUser
+    public void testGetCategories() throws Exception {
```
Comment (egorzhurov) [inline src/test/java/com/example/practice_modsen_shop/CategoryControllerTest.java]: remove public modifier

**Pull request closed without merging**